specs/DesignDiscussions/InputValidation.md

```diff
@@ -0,0 +1,503 @@
+            2. [Disabling Actions](#Disabling-Actions)
+    4. [Custom Rendering Error Messages](#Custom-Rendering-Error-Messages)
+    5. [Backwards Compatibility](#Backwards-Compatibility)
+    6. [Requirement Priorities](#Requirement-Priorities)
+2. [Annex](#Annex)
+    1. [Examples of invalid inputs](#Examples-of-invalid-inputs)
+    2. [Work for future iterations](#Work-for-future-iterations)
+        1. [Success Indicators](#Success-Indicators)
+        2. [Cross field validation](#Cross-field-validation)
+        3. [Future for validation and actions](#Future-for-validation-and-actions)
+            1. [Toggle and ShowCard Actions](#Toggle-and-ShowCard-Actions)
+            2. [OpenUrl](#OpenUrl)
+
+# Input Validation
+
+Input validation was originally spec'd in issue [#3081](https://github.com/microsoft/AdaptiveCards/issues/3081). This document covers the basics from that spec, and continues the discussion of additional concerns that it raised.
+
+## Schema Additions to Inputs for Validation
+
+``` json
+{
+    "type": "Input.Text",
+    "id": "username",
+    "isRequired": true,
+    "regex": "^\\w{5,50}$",
+    "errorMessage": "Username must be between 5 and 50 characters long."
+}
+```
+All `Input.*` types will have the `"isRequired"`, and `"errorMessage"` properties added to them. These properties allow a card author to specify whether or not an input is required, and what error message should be shown in case of invalid user input.
+
+Additionally, a `regex` property will be added to `Input.Text` to allow card authors to specify what regular expression defines valid input,
+
+## Validation Rendering
+
+By default, required inputs should be marked with a *. In cases where a label has been specified, the required indicator should be placed next to the input's label. Further discussion of label formatting can be found in [InputLabels.md](https://github.com/microsoft/AdaptiveCards/blob/master/specs/DesignDiscussions/InputLabels.md), tracked by [Issue #203](https://github.com/microsoft/AdaptiveCards/issues/203).
+
+![img](assets/InputValidation/Required.PNG)
+
+The error text and invalid indication will be rendered in the host's `attention` color.
+
+![img](assets/InputValidation/InputValidationExample.png)
+
+
+### Host side  styling
+
+As has been proposed, the error message is a string property which is rendered in the `attention` color by default, card hosts may need more control over how this messages and the visual representation of an error appear. As can be seen in the [Annex] there's a wide array on how error messages are represented in different websites. The table below details the proposed set of properties that host apps will be able to modify based on the examples located in the annex as well as if they are going to be modified through the host config or the native styling in each platform. It's important to note that some of this properties will act on the input element rather than the error message.
```

Comment:
>It's important to note that some of this properties will act on the input element rather than the error message. [](start = 578, length = 112)

Can you clarify this a bit? I assume "size" and "weight" do refer to the error message and that "spacing" refers to the space between the input and the error?